test/cpp/test_residency_profiling_capture_authority.cpp

```
@@ -0,0 +1,1436 @@
+        finish_thread_token_ = current_thread_token();
+        finish_observer_lifetime_ = current_thread_lifetime();
+        if (!active_ || token.opaque_id != token_.opaque_id) {
+            return failure_batch(
+                ProfilingIntervalSourceError::Failed,
+                after_event_watermark,
+                "dynamic finish token is inactive");
+        }
+        active_ = false;
+
+        if (scenario_.final_drain_event) {
+            append_frame(*scenario_.final_drain_event);
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)